src/modules/fastdelete/FastDelete.ShellExtension/FastDeleteForm.cs

```diff
@@ -0,0 +1,107 @@
+
+namespace FastDelete.ShellExtension
+{
+    public partial class FastDeleteForm : Form
+    {
+        public FastDeleteForm()
+        {
+            InitializeComponent();
+
+            // Gets the handle of the control to create the control on the VI thread. Invoking the Control.Handle get accessor forces the creation of the underlying window for the control.
+            // This is important, because the thread that instantiates the preview handler component and calls its constructor is a single-threaded apartment (STA) thread, but the thread that calls into the interface members later on is a multithreaded apartment (MTA) thread. Windows Forms controls are meant to run on STA threads.
+            // More details: https://docs.microsoft.com/en-us/archive/msdn-magazine/2007/january/windows-vista-and-office-writing-your-own-preview-handlers.
+            _ = this.Handle;
+        }
+
+        private Thread deleteThread = null;
+        private readonly CancellationTokenSource deleteCancelToken = new CancellationTokenSource();
+        public DirectoryInfo DirectoryToDelete { get; set; }
+
+        private void FastDeleteForm_Load(object sender, EventArgs e)
+        {
+            if (DirectoryToDelete == null)
+            {
+                throw new InvalidOperationException("DirectoryToDelete must be set before the form is shown");
+            }
+
+            mInstructionLabel.Text = $"Are you sure you want to delete \"{DirectoryToDelete.Name}\"?";
+        }
+
+        private void mCancelButton_Click(object sender, EventArgs e)
+        {
+            deleteCancelToken.Cancel();
+            deleteThread.Join();
+
+            DialogResult = DialogResult.Cancel;
+            Close();
+        }
+
+        private void mConfirmButton_Click(object sender, EventArgs e)
+        {
+            mConfirmButton.Visible = false;
+            mInstructionLabel.Visible = false;
+            mProgressBar.Visible = true;
+
+            deleteThread = new Thread(DeleteThreadProc);
+            deleteThread.SetApartmentState(ApartmentState.STA);
+            deleteThread.Start();
+        }
+
+        private static void DeleteThreadProc(object param)
+        {
+            FastDeleteForm form = (FastDeleteForm)param;
+            var token = form.deleteCancelToken.Token;
+
+            try
+            {
+                token.ThrowIfCancellationRequested();
+                ProcessStartInfo stage1proc = new ProcessStartInfo("C:\\Windows\\System32\\cmd.exe",
+                    "/c del /f/q/s *.* > nul");
+                stage1proc.WorkingDirectory = form.DirectoryToDelete.FullName;
+                Process.Start(stage1proc).WaitForExit();
+
+                token.ThrowIfCancellationRequested();
+                ProcessStartInfo stage2proc = new ProcessStartInfo("C:\\Windows\\System32\\cmd.exe",
+                    $"/c rmdir /s/q \"{form.DirectoryToDelete.FullName}\"");
+                Process.Start(stage2proc).WaitForExit();
```

Comment:
there is no need to spawn a cmd instance to do this, just use the file APIs directly